paasta_tools/drain_lib.py

```diff
@@ -100,7 +100,8 @@ class TestDrainMethod(DrainMethod):
     safe_to_kill_task_ids = set()
 
     def drain(self, task):
-        self.downed_task_ids.add(task.id)
+        if task.id not in self.safe_to_kill_task_ids:
```

Comment:
It's a bug, though it only manifests in a particular way:

 - a task is put in downed_task_ids by mark_arbitrary_task_as_safe_to_kill,
 - something (e.g. setup_marathon_job or native_mesos_scheduler) calls drain() again
 - then we call mark_arbitrary_task_as_safe_to_kill() again

In the existing itests, mark_arbitrary_task_as_safe_to_kill only gets called once, so this never happens.